grudge/array_context.py

```diff
@@ -239,7 +239,20 @@ def _dag_to_compiled_func(self, dict_of_named_arrays,
         self.actx._compile_trace_callback(self.f, "pre_find_distributed_partition",
                 dict_of_named_arrays)
 
-        distributed_partition = pt.find_distributed_partition(dict_of_named_arrays)
+        try:
+            distributed_partition = pt.find_distributed_partition(
+                                        dict_of_named_arrays)
+        except TypeError:
+            # https://github.com/inducer/pytato/pull/393 changes the
+            # function signature
+            # pylint: disable=too-many-function-args
+            distributed_partition = pt.find_distributed_partition(
+                                        # pylint-ignore-reason:
+                                        # '_BasePytatoArrayContext' has no
+                                        # 'mpi_communicator' member
+                                        # pylint: disable=no-member
+                                        self.actx.mpi_communicator,
+                                        dict_of_named_arrays)
```

Comment:
```suggestion
         # https://github.com/inducer/pytato/pull/393 changes the function signature
         try:
             # pylint: disable=too-many-function-args
             distributed_partition = pt.find_distributed_partition(
                 # pylint-ignore-reason:
                 # '_BasePytatoArrayContext' has no
                 # 'mpi_communicator' member
                 # pylint: disable=no-member
                 self.actx.mpi_communicator, dict_of_named_arrays)
         except TypeError as e:
             if "find_distributed_partition() takes 1 positional" in str(e):
                 distributed_partition = pt.find_distributed_partition(
                     dict_of_named_arrays)
             else:
                 raise
```
(So it doesn't catch every `TypeError`, just the relevant one.)

Reply:
Eh, sure. I can live with this.